Google drive support (WIP)

mduggan opened the pull request:
I need to run more testing to be happy that it all works correctly, but this is a first cut of Google Drive support that shares a lot of code with the Dropbox service.

Looking for feedback if you have some time to look at it.  The main TODO items are:
 * Store name -> id mapping information in the cache
 * Use file IDs where possible instead of path names (to avoid expensive lookups)

This addresses #82.

mduggan commented:
Made lots of improvements, I think it's basically ready to go now.  Please let me know what you think..

cpfair commented on tapiriik/local_settings.py.example:
```diff
@@ -7,6 +7,9 @@ DROPBOX_FULL_APP_SECRET = "####"
 DROPBOX_APP_KEY = "####"
 DROPBOX_APP_SECRET = "####"
 
+GOOGLEDRIVE_CLIENT_ID = "####"
```
Is this a special google drive credential? Or just a google app credential (thinking about when I get around to doing Google Fit).

mduggan replied:
Good question.  You can add multiple google APIs to the same project and make it a single confirmation from the user - so I guess it depends whether you want to tie the two of them together or not.  If you make them separate projects and allow people to hook up just one they'll have different client ID/client secret pairs.  The other option would just be to have one "Google" sync option and then make the fit/drive question user-configurable I guess.

mduggan replied:
By the way, you can create projects and configure the APIs they use at https://console.developers.google.com

natelowry commented:
`StorageServiceBase` will be awesome for adding OneDrive #143.  Any timeline for when this will get merged? :heart_eyes_cat:

NeverI commented:
It would be nice to merge this request!

dancingborg commented:
Any news on the progress of this pull request? I hope we can make the most of mduggan's great work.

patricksan commented:
Hey guys, any update on this one?

BullHorn7 commented:
Bump for interest!

randybb commented:
Definitely needed for people with many activities and without paid Dropbox.

piquet-h commented:
How far away is this from being merged?

bsiver commented:
Any update on this one?  Would really like to see Drive added as a sync option!

mat-ray commented:
Hello, any news on this merge?  Would be great to see Google Drive added.

bisraelsen commented:
Looks like this is abandoned?

mduggan commented:
I've rebased this and updated to work with the latest version of the dependencies.  It still works fine.  The `oauth2client` package has been deprecated so probably should be updated to something newer like `oauthlib`.

brisdaz commented:
What's the status? I'm new to GitHub and see that some have been "force-pushed" but it's not listed on the website?

mduggan commented:
In this case "force-pushed" means that I updated my copy of it with a new one and discarded my old attempt.  It's still up to @cpfair whether he wants to merge my changes - I think they are good to go, other than the fact that `oauth2client` is deprecated - it works fine though.
